tools/trace2syz/trace2syz/vmaCallHandlers.go

```diff
@@ -0,0 +1,357 @@
+package trace2syz
+
+import (
+	//"fmt"
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/prog"
+)
+
+const (
+	pageSize   = 4096
+	MapFixed   = "MAP_FIXED"
+	RemapFixed = "MREMAP_FIXED"
+)
+
+func ParseMemoryCall(ctx *Context) *prog.Call {
+	syzCall := ctx.CurrentSyzCall
+	straceCall := ctx.CurrentStraceCall
+	if straceCall.CallName == "mmap" {
+		return ParseMmap(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "mremap" {
+		return ParseMremap(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "msync" {
+		return ParseMsync(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "mprotect" {
+		return ParseMprotect(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "munmap" {
+		return ParseMunmap(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "madvise" {
+		return ParseMadvise(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "mlock" {
+		return ParseMlock(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "munlock" {
+		return ParseMunlock(syzCall.Meta, straceCall, ctx)
+	} else if straceCall.CallName == "shmat" {
+		return ParseShmat(syzCall.Meta, straceCall, ctx)
+	}
+	return nil
+}
+
+func ParseMmap(mmap *prog.Syscall, syscall *Syscall, ctx *Context) *prog.Call {
```

Comment:
things starting with capital letter are exported from package as public interface
most functions and types here should not be exported

Reply:
ack